src/tbb/task_dispatcher.h

```diff
@@ -299,8 +299,7 @@ d1::task* task_dispatcher::local_wait_for_all(d1::task* t, Waiter& waiter ) {
                 while (t != nullptr) {
                     assert_task_valid(t);
                     assert_pointer_valid</*alignment = */alignof(void*)>(ed.context);
-                    __TBB_ASSERT(ed.context->my_lifetime_state > d1::task_group_context::lifetime_state::locked &&
-                                 ed.context->my_lifetime_state < d1::task_group_context::lifetime_state::dying, nullptr);
+                    __TBB_ASSERT(ed.context->my_lifetime_state > d1::task_group_context::lifetime_state::locked, nullptr);
```

Comment:
Does it make sense to introduce `dead` state to track that we do not use destroyed context? E.g.
```suggestion
                    __TBB_ASSERT(ed.context->my_lifetime_state == d1::task_group_context::lifetime_state::bound &&
                                            ed.context->my_lifetime_state == d1::task_group_context::lifetime_state::isolated, nullptr);
```